[blocker] CLOUDSTACK-9452: add python-argparse dependency on el6,7 rpms

rhtyd opened the pull request:
The patchviasocket script was rewritten in Python from PR #1533 and made
assumptions that Python 2.7 would be available. In case of CentOS, python 2.7
may not be available or installed. This change ensures that python-argparse
is installed which is used by this script.

/cc @wido @sverrirab @karuturi @jburwell 

@blueorangutan package

blueorangutan commented:
@rhtyd a Jenkins job has been kicked to build packages. I'll keep you posted as I make progress.

sverrirab commented:
+1 from me (for what it's worth)

blueorangutan commented:
Packaging result: ✔centos6 ✔centos7 ✔debian repo: http://packages.shapeblue.com/cloudstack/pr/1634
Job ID-77

jburwell commented:
LGTM for code

rhtyd commented:
@karuturi @jburwell can we merge this, this is only packaging related fix. Using above packages, installation confirmed that argparse was installed on el6/el7 kvm hosts.

rhtyd commented:
@blueorangutan test

blueorangutan commented:
@rhtyd a Trillian-Jenkins test job (centos6 mgmt + vmware-55u3) has been kicked to run smoke tests against packages at http://packages.shapeblue.com/cloudstack/pr/1634

blueorangutan commented:
Trillian test result (trillian-pr1634-33-vmware-55u3-cs49): 
Test completed. 39 look ok, 13 have errors
Test | Result | Time
--- | --- | ---
test_DeployVmAntiAffinityGroup_in_project | Success | 242.223
test_create_pvlan_network | Success | 5.174
test_DeployVmAntiAffinityGroup | Success | 222.027
test_dedicatePublicIpRange | `Error` | 0.260
test_deploy_vgpu_enabled_vm | Skipped | 0.003
test_deploy_vm_from_iso | Success | 373.043
test_01_scale_vm | Skipped | 66.406
test_00_deploy_vm_root_resize | Success | 6.358
test_01_deploy_vm_root_resize | Success | 6.338
test_02_deploy_vm_root_resize | Success | 6.345
test_01_router_internal_basic | Success | 0.494
test_02_router_internal_adv | Success | 0.903
test_03_restart_network_cleanup | Success | 140.928
test_04_restart_network_wo_cleanup | Success | 5.591
test_05_router_basic | Success | 0.032
test_06_router_advanced | Success | 0.042
test_07_stop_router | Success | 25.274
test_08_start_router | Success | 120.878
test_09_reboot_router | Success | 140.902
test_deployvm_firstfit | Success | 206.794
test_deployvm_userconcentrated | Success | 96.038
test_deployvm_userdispersing | Success | 60.780
test_deployvm_userdata | Success | 151.369
test_deployvm_userdata_post | Success | 25.429
test_01_sys_vm_start | Success | 0.122
test_02_sys_template_ready | Success | 0.098
test_01_create_disk_offering | Success | 0.127
test_02_create_sparse_type_disk_offering | Success | 0.065
test_04_create_fat_type_disk_offering | Success | 0.067
test_02_edit_disk_offering | Success | 0.058
test_03_delete_disk_offering | Success | 0.037
test_default_role_deletion | Success | 6.364
test_role_account_acls | Success | 7.961
test_role_account_acls_multiple_mgmt_servers | Success | 7.946
test_role_inuse_deletion | Success | 6.231
test_role_lifecycle_create | Success | 6.400
test_role_lifecycle_delete | Success | 11.345
test_role_lifecycle_list | Success | 6.348
test_role_lifecycle_update | Success | 11.434
test_role_lifecycle_update_role_inuse | Success | 6.291
test_rolepermission_lifecycle_concurrent_updates | Success | 6.465
test_rolepermission_lifecycle_create | Success | 6.343
test_rolepermission_lifecycle_delete | Success | 6.226
test_rolepermission_lifecycle_list | Success | 6.391
test_rolepermission_lifecycle_update | Success | 6.513
test_01_create_service_offering | Success | 0.102
test_02_edit_service_offering | Success | 0.078
test_03_delete_service_offering | Success | 0.045
test_04_change_offering_small | Success | 96.948
test_UpdateConfigParamWithScope | Success | 0.145
ContextSuite context=TestDedicateGuestVlanRange>:setup | `Error` | 0.000
test_01_snapshot_root_disk | Success | 157.389
test_01_internallb_roundrobin_1VPC_3VM_HTTP_port80 | `Failure` | 473.466
test_02_internallb_roundrobin_1RVPC_3VM_HTTP_port80 | `Failure` | 642.115
test_03_vpc_internallb_haproxy_stats_on_all_interfaces | `Error` | 355.933
test_04_rvpc_internallb_haproxy_stats_on_all_interfaces | `Error` | 487.776
test_01_list_sec_storage_vm | Success | 0.133
test_02_list_cpvm_vm | Success | 0.106
test_03_ssvm_internals | Success | 3.458
test_04_cpvm_internals | Success | 1.075
test_05_stop_ssvm | Success | 168.743
test_06_stop_cpvm | Success | 201.889
test_07_reboot_ssvm | Success | 158.628
test_08_reboot_cpvm | Success | 156.591
test_09_destroy_ssvm | Success | 269.128
test_10_destroy_cpvm | Success | 266.909
test_01_create_iso | Success | 66.612
test_02_edit_iso | Success | 0.083
test_03_delete_iso | Success | 95.198
test_04_extract_Iso | Success | 5.164
test_05_iso_permissions | Success | 0.050
test_06_copy_iso | Skipped | 0.000
test_07_list_default_iso | Success | 0.037
test_01_list_volumes | Success | 0.021
test_02_list_templates | Success | 0.029
test_03_list_snapshots | Success | 0.064
test_static_role_account_acls | Skipped | 0.020
test_01_create_lb_rule_src_nat | Success | 208.166
test_02_create_lb_rule_non_nat | Success | 207.423
test_assign_and_removal_lb | Success | 148.813
login_test_saml_user | Success | 21.747
test_01_create_template | Success | 135.939
test_CreateTemplateWithDuplicateName | Success | 221.472
test_02_edit_template | Success | 90.116
test_03_delete_template | Success | 5.123
test_04_extract_template | Success | 10.319
test_05_template_permissions | Success | 0.047
test_06_copy_template | Skipped | 0.000
test_07_list_public_templates | Success | 0.029
test_08_list_system_templates | Success | 0.068
test_nic_secondaryip_add_remove | Success | 222.996
test_network_acl | Success | 156.582
test_delete_account | Success | 288.136
test_01_port_fwd_on_src_nat | Success | 111.745
test_02_port_fwd_on_non_src_nat | Success | 55.752
test_public_ip_admin_account | Success | 40.300
test_public_ip_user_account | Success | 10.288
test_reboot_router | Success | 609.980
test_releaseIP | Success | 278.713
test_network_rules_acquired_public_ip_1_static_nat_rule | Success | 125.334
test_network_rules_acquired_public_ip_2_nat_rule | Success | 61.565
test_network_rules_acquired_public_ip_3_Load_Balancer_Rule | Success | 76.872
test_vm_nic_adapter_vmxnet3 | Skipped | 0.001
test_advZoneVirtualRouter | Success | 0.019
test_deploy_vm | Success | 0.023
test_deploy_vm_multiple | Success | 222.726
test_01_stop_vm | Success | 5.101
test_02_start_vm | Success | 20.211
test_03_reboot_vm | Success | 5.131
test_06_destroy_vm | Success | 10.126
test_07_restore_vm | Success | 0.090
test_08_migrate_vm | Success | 76.035
test_09_expunge_vm | Success | 125.250
test_10_attachAndDetach_iso | Success | 71.856
test_01_nic | Success | 671.990
test_extendPhysicalNetworkVlan | `Error` | 0.031
test_01_test_vm_volume_snapshot | `Error` | 146.123
test_01_create_vm_snapshots | Success | 161.769
test_02_revert_vm_snapshots | Success | 195.950
test_03_delete_vm_snapshots | Success | 275.255
test_oobm_background_powerstate_sync | Success | 15.838
test_oobm_configure_default_driver | Success | 0.094
test_oobm_configure_invalid_driver | Success | 0.082
test_oobm_disable_feature_invalid | Success | 0.104
test_oobm_disable_feature_valid | Success | 5.172
test_oobm_enable_feature_invalid | Success | 0.098
test_oobm_enable_feature_valid | Success | 5.176
test_oobm_enabledisable_across_clusterzones | Success | 53.525
test_oobm_issue_power_cycle | Success | 15.573
test_oobm_issue_power_off | Success | 15.576
test_oobm_issue_power_on | Success | 15.583
test_oobm_issue_power_reset | Success | 15.576
test_oobm_issue_power_soft | Success | 15.571
test_oobm_issue_power_status | Success | 10.546
test_oobm_multiple_mgmt_server_ownership | Success | 12.545
test_oobm_zchange_password | Success | 20.588
test_UpdateStorageOverProvisioningFactor | Success | 0.140
test_01_create_volume | Success | 513.042
test_02_attach_volume | Success | 53.973
test_03_download_attached_volume | Success | 20.340
test_04_delete_attached_volume | Success | 20.252
test_05_detach_volume | Success | 110.314
test_06_download_detached_volume | `Error` | 45.471
test_07_resize_fail | Skipped | 15.316
test_08_resize_volume | Skipped | 5.118
test_09_delete_detached_volume | Success | 30.832
test_isolate_network_password_server | `Failure` | 168.605
test_createPortablePublicIPAcquire | Success | 15.577
test_createPortablePublicIPRange | Success | 15.310
test_01_create_redundant_VPC_2tiers_4VMs_4IPs_4PF_ACL | `Failure` | 559.905
test_02_redundant_VPC_default_routes | Success | 729.590
test_03_create_redundant_VPC_1tier_2VMs_2IPs_2PF_ACL_reboot_routers | Success | 512.650
test_04_rvpc_network_garbage_collector_nics | Success | 846.858
test_05_rvpc_multi_tiers | Success | 721.879
test_01_primary_storage_iscsi | Skipped | 0.033
test_01_primary_storage_nfs | Success | 37.098
test_01_vpc_privategw_acl | `Failure` | 111.974
test_02_vpc_privategw_static_routes | `Failure` | 217.688
test_03_vpc_privategw_restart_vpc_cleanup | `Failure` | 217.559
test_04_rvpc_privategw_static_routes | `Failure` | 403.800
test_01_VPC_nics_after_destroy | Success | 746.848
test_02_VPC_default_routes | Success | 410.326
test_createRegion | Success | 0.039
test_01_reset_vm_on_reboot | Success | 25.307
test_01_updatevolumedetail | Success | 0.104
test_router_dhcphosts | `Failure` | 168.753
ContextSuite context=TestRouterDHCPHosts>:teardown | `Error` | 189.048
test_02_routervm_iptables_policies | `Failure` | 211.798
test_01_single_VPC_iptables_policies | `Error` | 121.100
test_01_isolate_network_FW_PF_default_routes_egress_true | Success | 283.827
test_02_isolate_network_FW_PF_default_routes_egress_false | Success | 424.548
test_01_RVR_Network_FW_PF_SSH_default_routes_egress_true | `Failure` | 415.926
test_02_RVR_Network_FW_PF_SSH_default_routes_egress_false | `Failure` | 437.656
test_03_RVR_Network_check_router_state | Success | 366.057
test_01_redundant_vpc_site2site_vpn | `Error` | 738.662
test_01_vpc_remote_access_vpn | `Error` | 0.074
test_01_vpc_site2site_vpn | `Error` | 507.094
Trillian env - trillian-pr1634-33-vmware-55u3-cs49, Job ID 33
Marvin logs at: http://packages.shapeblue.com/cloudstack/pr/1634/trillian/trillian-pr1634-33-vmware-55u3-cs49

jburwell commented:
@rhtyd the test results from blueorangutan are looking pretty good.  A couple of observations:

1. The markdown for the table seems to be invalid as it is not rendering to a table
2. Would it be possible to sort the rows in Failure, Skip, Success order?  We are most interested in failures and skips.  Placing them at the top will speed visual scanning up a bit.
3.  Include a summary of the environment used to run the tests

rhtyd commented:
@jburwell  thanks

- I've fixed the markdown issues -- see this for example -- https://github.com/apache/cloudstack/pull/1638#issuecomment-241644475
- I'll fix the sorting issues, to sort status column
- I've added summary at the bottom of the table, noting the hypervisor, mgmt server os, zone type and link to the marvin logs for anyone to inspect further

rhtyd commented:
We've enough LGTMs and test result (with vmware), I'll fire another Trillian job with KVM.

blueorangutan commented:
Trillian test result (trillian-pr1634-49-kvm-centos68-cs49): 
Test completed. 45 look ok, 7 have errors


Test | Result | Time (s)
--- | --- | ---
test_04_rvpc_privategw_static_routes | `Failure` | 173.220
test_03_vpc_privategw_restart_vpc_cleanup | `Failure` | 87.754
test_02_vpc_privategw_static_routes | `Failure` | 82.476
test_01_vpc_privategw_acl | `Failure` | 47.131
test_extendPhysicalNetworkVlan | `Error` | 0.039
test_dedicatePublicIpRange | `Error` | 0.322
test_01_vpc_remote_access_vpn | `Error` | 0.092
test_01_snapshot_root_disk | `Error` | 0.059
ContextSuite context=TestListIdsParams>:setup | `Error` | 0.000
ContextSuite context=TestDedicateGuestVlanRange>:setup | `Error` | 0.000
test_router_dhcphosts | Success | 291.471
test_rolepermission_lifecycle_update | Success | 7.183
test_rolepermission_lifecycle_list | Success | 6.939
test_rolepermission_lifecycle_delete | Success | 6.798
test_rolepermission_lifecycle_create | Success | 6.862
test_rolepermission_lifecycle_concurrent_updates | Success | 6.969
test_role_lifecycle_update_role_inuse | Success | 7.025
test_role_lifecycle_update | Success | 11.926
test_role_lifecycle_list | Success | 6.845
test_role_lifecycle_delete | Success | 11.877
test_role_lifecycle_create | Success | 6.825
test_role_inuse_deletion | Success | 6.814
test_role_account_acls_multiple_mgmt_servers | Success | 8.997
test_role_account_acls | Success | 9.337
test_releaseIP | Success | 249.057
test_reboot_router | Success | 420.117
test_public_ip_user_account | Success | 10.306
test_public_ip_admin_account | Success | 40.320
test_oobm_zchange_password | Success | 20.649
test_oobm_multiple_mgmt_server_ownership | Success | 18.590
test_oobm_issue_power_status | Success | 10.563
test_oobm_issue_power_soft | Success | 15.583
test_oobm_issue_power_reset | Success | 15.595
test_oobm_issue_power_on | Success | 15.583
test_oobm_issue_power_off | Success | 15.592
test_oobm_issue_power_cycle | Success | 15.599
test_oobm_enabledisable_across_clusterzones | Success | 58.881
test_oobm_enable_feature_valid | Success | 5.220
test_oobm_enable_feature_invalid | Success | 0.135
test_oobm_disable_feature_valid | Success | 5.256
test_oobm_disable_feature_invalid | Success | 0.155
test_oobm_configure_invalid_driver | Success | 0.105
test_oobm_configure_default_driver | Success | 0.118
test_oobm_background_powerstate_sync | Success | 17.821
test_nic_secondaryip_add_remove | Success | 194.173
test_network_rules_acquired_public_ip_3_Load_Balancer_Rule | Success | 67.156
test_network_rules_acquired_public_ip_2_nat_rule | Success | 61.934
test_network_rules_acquired_public_ip_1_static_nat_rule | Success | 154.560
test_network_acl | Success | 66.274
test_isolate_network_password_server | Success | 93.446
test_deployvm_userdispersing | Success | 20.671
test_deployvm_userdata_post | Success | 10.473
test_deployvm_userdata | Success | 121.267
test_deployvm_userconcentrated | Success | 86.630
test_deployvm_firstfit | Success | 55.723
test_deploy_vm_multiple | Success | 258.971
test_deploy_vm_from_iso | Success | 428.730
test_deploy_vm | Success | 0.027
test_delete_account | Success | 259.013
test_default_role_deletion | Success | 6.919
test_create_pvlan_network | Success | 5.232
test_createRegion | Success | 0.052
test_createPortablePublicIPRange | Success | 15.427
test_createPortablePublicIPAcquire | Success | 15.699
test_assign_and_removal_lb | Success | 134.011
test_advZoneVirtualRouter | Success | 0.022
test_UpdateStorageOverProvisioningFactor | Success | 0.154
test_UpdateConfigParamWithScope | Success | 0.157
test_DeployVmAntiAffinityGroup_in_project | Success | 197.434
test_DeployVmAntiAffinityGroup | Success | 71.075
test_CreateTemplateWithDuplicateName | Success | 121.145
test_10_destroy_cpvm | Success | 191.835
test_10_attachAndDetach_iso | Success | 71.829
test_09_reboot_router | Success | 45.381
test_09_expunge_vm | Success | 125.174
test_09_destroy_ssvm | Success | 163.950
test_09_delete_detached_volume | Success | 15.699
test_08_start_router | Success | 30.316
test_08_resize_volume | Success | 15.476
test_08_reboot_cpvm | Success | 131.901
test_08_migrate_vm | Success | 41.761
test_08_list_system_templates | Success | 0.031
test_07_stop_router | Success | 10.164
test_07_restore_vm | Success | 0.132
test_07_resize_fail | Success | 20.547
test_07_reboot_ssvm | Success | 133.872
test_07_list_public_templates | Success | 0.035
test_07_list_default_iso | Success | 0.063
test_06_stop_cpvm | Success | 131.868
test_06_router_advanced | Success | 0.059
test_06_download_detached_volume | Success | 15.344
test_06_destroy_vm | Success | 126.011
test_05_template_permissions | Success | 0.069
test_05_stop_ssvm | Success | 134.130
test_05_rvpc_multi_tiers | Success | 493.002
test_05_router_basic | Success | 0.064
test_05_iso_permissions | Success | 0.070
test_05_detach_volume | Success | 100.333
test_04_rvpc_network_garbage_collector_nics | Success | 848.811
test_04_rvpc_internallb_haproxy_stats_on_all_interfaces | Success | 253.859
test_04_restart_network_wo_cleanup | Success | 5.808
test_04_extract_template | Success | 5.401
test_04_extract_Iso | Success | 5.173
test_04_delete_attached_volume | Success | 10.259
test_04_create_fat_type_disk_offering | Success | 0.081
test_04_cpvm_internals | Success | 1.336
test_04_change_offering_small | Success | 239.157
test_03_vpc_internallb_haproxy_stats_on_all_interfaces | Success | 198.983
test_03_ssvm_internals | Success | 3.637
test_03_restart_network_cleanup | Success | 65.585
test_03_reboot_vm | Success | 125.935
test_03_download_attached_volume | Success | 15.382
test_03_delete_template | Success | 5.133
test_03_delete_service_offering | Success | 0.042
test_03_delete_iso | Success | 95.231
test_03_delete_disk_offering | Success | 0.045
test_03_create_redundant_VPC_1tier_2VMs_2IPs_2PF_ACL_reboot_routers | Success | 508.708
test_03_RVR_Network_check_router_state | Success | 265.179
test_02_sys_template_ready | Success | 0.127
test_02_start_vm | Success | 10.229
test_02_routervm_iptables_policies | Success | 187.674
test_02_router_internal_adv | Success | 1.048
test_02_redundant_VPC_default_routes | Success | 550.962
test_02_port_fwd_on_non_src_nat | Success | 55.739
test_02_list_cpvm_vm | Success | 0.139
test_02_isolate_network_FW_PF_default_routes_egress_false | Success | 253.682
test_02_internallb_roundrobin_1RVPC_3VM_HTTP_port80 | Success | 526.427
test_02_edit_template | Success | 90.208
test_02_edit_service_offering | Success | 0.088
test_02_edit_iso | Success | 0.092
test_02_edit_disk_offering | Success | 0.066
test_02_deploy_vm_root_resize | Success | 6.991
test_02_create_sparse_type_disk_offering | Success | 0.079
test_02_create_lb_rule_non_nat | Success | 187.690
test_02_attach_volume | Success | 74.832
test_02_VPC_default_routes | Success | 333.557
test_02_RVR_Network_FW_PF_SSH_default_routes_egress_false | Success | 329.590
test_01_vpc_site2site_vpn | Success | 165.492
test_01_updatevolumedetail | Success | 0.104
test_01_sys_vm_start | Success | 0.178
test_01_stop_vm | Success | 40.425
test_01_single_VPC_iptables_policies | Success | 319.157
test_01_router_internal_basic | Success | 0.610
test_01_reset_vm_on_reboot | Success | 130.985
test_01_redundant_vpc_site2site_vpn | Success | 267.285
test_01_primary_storage_nfs | Success | 35.964
test_01_port_fwd_on_src_nat | Success | 141.937
test_01_nic | Success | 521.193
test_01_list_sec_storage_vm | Success | 0.177
test_01_isolate_network_FW_PF_default_routes_egress_true | Success | 238.678
test_01_internallb_roundrobin_1VPC_3VM_HTTP_port80 | Success | 446.704
test_01_deploy_vm_root_resize | Success | 6.933
test_01_create_volume | Success | 861.863
test_01_create_template | Success | 70.970
test_01_create_service_offering | Success | 0.124
test_01_create_redundant_VPC_2tiers_4VMs_4IPs_4PF_ACL | Success | 1062.368
test_01_create_lb_rule_src_nat | Success | 239.855
test_01_create_iso | Success | 66.895
test_01_create_disk_offering | Success | 0.126
test_01_VPC_nics_after_destroy | Success | 574.847
test_01_RVR_Network_FW_PF_SSH_default_routes_egress_true | Success | 318.416
test_00_deploy_vm_root_resize | Success | 253.906
login_test_saml_user | Success | 24.513
test_vm_nic_adapter_vmxnet3 | Skipped | 0.000
test_static_role_account_acls | Skipped | 0.021
test_deploy_vgpu_enabled_vm | Skipped | 0.005
test_06_copy_template | Skipped | 0.000
test_06_copy_iso | Skipped | 0.000
test_03_delete_vm_snapshots | Skipped | 0.000
test_02_revert_vm_snapshots | Skipped | 0.000
test_01_test_vm_volume_snapshot | Skipped | 0.000
test_01_scale_vm | Skipped | 0.000
test_01_primary_storage_iscsi | Skipped | 0.036
test_01_create_vm_snapshots | Skipped | 0.000


Trillian env - trillian-pr1634-49-kvm-centos68-cs49, Job ID 49
Hypervisor: kvm-centos68 (x2), Advanced Zone
Mgmt host os - 6
Marvin logs at: http://packages.shapeblue.com/cloudstack/pr/1634/trillian/trillian-pr1634-49-kvm-centos68-cs49

rhtyd commented:
Merging based on lgtms and test results (the failures are due to env issues, and failing tests not related to the pkging fix).
